test/functional/test_framework/socks5.py

```diff
@@ -117,6 +128,33 @@ def handle(self):
             cmdin = Socks5Command(cmd, atyp, addr, port, username, password)
             self.serv.queue.put(cmdin)
             logger.debug('Proxy: %s', cmdin)
+
+            num_destinations = len(self.serv.conf.destinations)
+            if self.serv.conf.destinations_used < num_destinations:
+                dest = self.serv.conf.destinations[self.serv.conf.destinations_used]
+                dest["requested_to_addr"] = addr.decode("utf-8")
+                self.serv.conf.destinations_used += 1
+                with socket.create_connection((dest["to_addr"], dest["to_port"])) as conn_to:
+                    self.conn.setblocking(False)
+                    conn_to.setblocking(False)
+                    sockets = [self.conn, conn_to]
+                    done = False
+                    while not done:
+                        rlist, _, xlist = select.select(sockets, [], sockets)
+                        if len(xlist) > 0:
+                            raise IOError("Exceptional condition on socket")
+                        for s in rlist:
+                            data = s.recv(4096)
```

Comment:
No, I just picked some value that looks reasonable, anything would work. For example `1` or `5` would still work but does not look reasonable because it will cause a lot of iteration loops.